### PR TITLE
fix: surface unparseable edge conditions as DIP010 (Closes #98)

### LINKED ISSUE #98: validate/lint/doctor pass on workflows with invalid edge conditions (error from EnsureConditionsParsed is discarded)
## Summary

`validator.Lint()` **silently discards** the error from `simulate.EnsureConditionsParsed()`, and that function **returns on the first unparseable edge condition**. The combination means a workflow with an invalid `when ...` condition reports **"validation passed" / doctor Grade A**, yet hard-fails at `dippin simulate` (and at runtime). Worse, because the parse loop stops at the first bad edge, every edge *after* it keeps `Condition.Parsed == nil`, silently disabling the lint checks that depend on the AST (DIP101/102/103/120) for the remainder of the graph.

This is a false-confidence bug: `dippin validate`/`lint`/`doctor` are exactly the commands an author runs to gain confidence their `.dip` is correct, and here they greenlight a file that cannot execute.

## Root cause

`validator/lint.go:20`:

```go
func Lint(w *ir.Workflow) Result {
	// Ensure condition ASTs are populated ...
	_ = simulate.EnsureConditionsParsed(w)   // <-- error discarded
	...
}
```

`simulate/condition.go:269` returns on the first failure, so the loop never reaches later edges:

```go
func EnsureConditionsParsed(w *ir.Workflow) error {
	for _, e := range w.Edges {
		if err := ensureEdgeConditionParsed(e); err != nil {
			return err   // <-- bails; remaining edges keep Parsed == nil
		}
	}
	...
}
```

Downstream, `lintConditionalReachability`/`lintDefaultEdge`/`lintOverlappingConditions` only reason about edges whose `Condition.Parsed != nil`, so a `nil` AST reads as "nothing to check" rather than "couldn't parse." The bad condition itself produces **no diagnostic at all**.

## Reproduction

A minimal workflow with an invalid edge operator (e.g. the `marker_grep`-as-edge-keyword mistake, which is easy to make since `marker_grep` is a real *tool-node* field):

```dippin
workflow Repro
  goal: "repro"
  start: A
  exit: Z

  tool A
    command: "echo ok"
  agent Z
    prompt: "done"

  edges
    A -> Z when marker_grep "^ok"
```

- `dippin validate` / `dippin lint` / `dippin doctor` → **pass** (doctor Grade A).
- `dippin simulate` → `edge A -> Z: invalid condition "marker_grep \"^ok\"": unknown operator "^ok"`.

(Observed during a design review of an external workflow; the entire ~22-edge routing spine was invalid yet validated clean.)

## Impact

- Authors treat "validation passed" as proof of routing correctness; it isn't.
- The first invalid condition masks DIP101/102/103/120 for **all subsequent edges**, so even unrelated real warnings go silent.
- Invalid conditions are easy to write (wrong operator, `marker_grep`/regex confusion, `ctx.*` typos in the operator position) and currently surface only at `simulate`/runtime.

## Proposed fix

Surface the parse failure as a **structural diagnostic** instead of swallowing it. Options, roughly in order of preference:

1. **New structural code (e.g. DIP00x "unparseable edge condition").** Have `EnsureConditionsParsed` accumulate *all* failures (don't early-return), and have `Lint()`/`Validate()` emit one diagnostic per bad condition with the edge and the parser error. This both reports the real problem and lets parsing continue so later edges still get linted.
2. **Minimal version:** keep the single-error behavior but stop discarding it — convert the returned error into a diagnostic in `Lint()` (and surface it from `dippin validate`/`check`). Less ideal because it still stops at the first bad edge.
3. Independently, **add `dippin simulate` (or a condition-parse pass) to `just check` / CI** so condition validity is gated even if (1)/(2) slip.

I'd lean toward (1): a dedicated code keeps it visible in the spec/explanations, and accumulating errors removes the cascade-masking of DIP101/102/103/120.

Happy to put up the PR if the approach sounds right.

## Refs

- `validator/lint.go:20`
- `simulate/condition.go:269` (`EnsureConditionsParsed`), `:284` (`ensureEdgeConditionParsed`), `:288` (`ParseCondition`)
- Discovered while reviewing `2389-research/pipelines#40`.

### PR DESCRIPTION
**Pull request by clintecker**
## Problem

An invalid edge `when` condition — e.g. `A -> Z when marker_grep "^ok"` (easy to write, since `marker_grep` is a real *tool-node* field) — passed `validate`/`lint`/`check`/`doctor` **silently**, yet hard-failed at `dippin simulate` and at runtime:

```
edge A -> Z: invalid condition "marker_grep \"^ok\"": unknown operator "^ok"
```

`validator.Lint()` discarded the error from `simulate.EnsureConditionsParsed()` (`validator/lint.go:20`), which **early-returned on the first** unparseable edge. So the bad condition produced no diagnostic, and every edge *after* it kept `Condition.Parsed == nil`, silently disabling the AST-dependent lints for the rest of the graph. These are exactly the commands an author runs to gain confidence a `.dip` is correct.

## Fix

- **New structural code DIP010** (`SeverityError`), emitted from `validator.Validate()` — which is run by **every** command path (validate/lint/check/watch/simulate/pack/wasm/doctor), so one check is caught everywhere. `validator/validate_conditions.go` parses each edge via the already-exported, wasm-safe `simulate.ParseCondition`, populating `Condition.Parsed` for every parseable edge and collecting **one failure per bad edge** (accumulate-all — parsing no longer stops at the first failure, so later edges keep getting linted).
- **`EnsureConditionsParsed` and its 6 callers are untouched** — no `simulate` API change, `wasm` stays green.
- **`Lint()`** now populates edge ASTs accumulate-all too, so a standalone lint pass is also un-masked.
- **`dippin doctor`** now exits non-zero when the report contains errors (it previously always exited 0), so it no longer greenlights a workflow that cannot execute. The report still renders in full.

## Corrected premise

The issue named DIP101/102/103/120 as the masked lints. Empirically (verified in review) only the `.Parsed`-dependent lints — **DIP103/120/121/122** — are masked; DIP101/102 fire on condition *presence* alone. The cascade regression test therefore asserts a downstream **DIP120** survives an earlier bad edge (a fixture that genuinely discriminates the fix).

## Scope

Edge conditions only. `manager_loop` stop/steer node conditions have the same swallowed-error class but no lint reads their `.Parsed` today — noted as a separate follow-up.

## Tests

All real-parser-driven (no hand-built IR with pre-set `.Parsed`):
- repro → DIP010 fires, `validate`/`check`/`doctor` exit non-zero
- **cascade** → DIP120 still fires on a later edge despite an earlier unparseable one
- one DIP010 per bad edge (count == 2); valid conditions clean; `parseEdgeConditions` unit; manager_loop edges-only scope boundary
- CLI integration (stream-correct: validate→stderr, check→stdout); `TestCmdDoctor_WithSuggestions` updated to `ExitError`

## Notes

- Catalog counts / structural range bumped to **DIP001–DIP010 / 56 codes**; generated spec regenerated. Reviewed via a 5-expert design panel + adversarial code-review pass.
- After merge this batches into a minor bump — not tagging without the maintainer's go-ahead.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/2389-research/codesmith/dippin-lang/pr/103"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783525014&installation_id=131176874&pr_number=103&repository=2389-research%2Fdippin-lang&return_to=https%3A%2F%2Fgithub.com%2F2389-research%2Fdippin-lang%2Fpull%2F103&signature=49601d50752864f8bef616e36d423f40d1925dd768c99076a7ce3b119d7fa3f1"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added DIP010 (unparseable edge condition); each bad edge emits one error.
  * `doctor` now exits non-zero when the health report contains errors while still rendering the full report.

* **Documentation**
  * Updated CLI/help, site docs, and changelog to reflect structural validation DIP001–DIP010, lint warnings DIP101–DIP146, and updated diagnostic totals (56).

### CODE REVIEW

File: validator/validate_conditions.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// (DIP010). It is separate from validate.go so the dependency on the simulate
+// package — the only cross-package import the validator's structural checks
+// have — stays isolated to a single file. simulate.ParseCondition is pure
+// string→AST (no filesystem), so this remains wasm-safe.
+
+// edgeNeedsParse reports whether an edge's condition still needs parsing.
+// Mirrors the guard in simulate.ensureEdgeConditionParsed: skip conditions that
+// are absent, already parsed, or have no raw text.
+func edgeNeedsParse(e *ir.Edge) bool {
+	return e.Condition != nil && e.Condition.Parsed == nil && e.Condition.Raw != ""
```
[the remark this answers is not in the document]
Comment (clintecker) [reply validator/validate_conditions.go:20]: Verified against the parser and runtime: both `A -> B when` (empty) and `A -> B when label: ok` (label-only — `readConditionRaw` stops at the `label:` keyword, so `Raw == ""`) produce `Condition{Raw: ""}`, and crucially **both `dippin validate` and `dippin simulate` succeed (exit 0), traversing the edge as unconditional**. So the validator and runtime *agree* here — there is no validate-vs-runtime gap. DIP010 specifically targets conditions that pass validation but **fail at simulate** (#98). The `Raw == ""` skip in `edgeNeedsParse` mirrors `simulate.ensureEdgeConditionParsed` exactly; flagging empty `when` would make the validator diverge from the runtime (a false positive on an edge that runs fine), and is out of scope for #98. A separate advisory lint for "likely-incomplete `when`" could be filed if that footgun is worth catching.

File: cmd/dippin/generated-spec.md [hunk 2]
```diff
@@ -189,9 +189,9 @@ Workflow: author and lint as `.dip`; package with `dippin pack` for distribution
 
 ## Diagnostic Code Summary
 
-55 diagnostic codes across two categories:
+56 diagnostic codes across two categories:
 
-- **DIP001–DIP009** (errors): start/exit missing, unknown refs, unreachable nodes, cycles, duplicates, parallel/fan_in mismatch
+- **DIP001–DIP010** (errors): start/exit missing, unknown refs, unreachable nodes, cycles, duplicates, parallel/fan_in mismatch, unparseable edge conditions
```
[the remark this answers is not in the document]
Comment (clintecker) [reply cmd/dippin/generated-spec.md:194]: Stale info — the committed `cmd/dippin/generated-spec.md` is in sync: re-running `./scripts/gen-spec.sh` produces no diff, and the CI **Test** job (which runs `releasecheck.TestGeneratedSpecIsCurrentWithGenerator`) is **green** on this PR. The pre-commit hook auto-regenerates + runs releasecheck on every commit. No action needed.

File: docs/cli.md [hunk 3]
```diff
@@ -124,15 +124,15 @@ error[DIP003]: unknown node reference "InterpretX" in edge
...
+Run both structural validation and semantic linting (DIP001–DIP010 + DIP101–DIP145).
 
 ```bash
 dippin lint [--extra-models <spec>] <file>
 ```
 
 **Input**: `.dip` or `.dot` file
 
-**Checks**: All 54 diagnostic rules. Errors (DIP001–DIP009) cause exit code 1. Warnings (DIP101–DIP145) are reported but don't affect the exit code.
+**Checks**: All 55 diagnostic rules. Errors (DIP001–DIP010) cause exit code 1. Warnings (DIP101–DIP145) are reported but don't affect the exit code.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply docs/cli.md]: The `DIP101–DIP145` → `DIP101–DIP146` staleness is pre-existing drift from #99 (DIP146 shipped before this branch) and spans ~8 files (architecture.md, integration.md, editor-setup.md, llms-full.txt, …), not just this line. Fixing it piecemeal here would be inconsistent and is not traceable to #98 — better as a dedicated docs-sync. The "55 diagnostic rules" count is intentional: 56 codes minus the reserved DIP138 = 55 firing/documented rules, matching the "55 documented sections" framing on the validation page. Leaving as-is for this PR.

File: site/content/validation.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ subtitle: "40 diagnostic checks — 9 structural errors and 31 semantic warnings
 
 Dippin provides two levels of analysis:
 
-**Structural validation** (DIP001-DIP009): Errors that must be fixed. A workflow with any of these cannot execute. Run with `dippin validate`.
+**Structural validation** (DIP001-DIP010): Errors that must be fixed. A workflow with any of these cannot execute. Run with `dippin validate`.
 
 **Semantic linting** (DIP101-DIP134): Warnings that flag likely bugs or questionable patterns. They don't block execution but should be reviewed. Run with `dippin lint` for both levels.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply site/content/validation.md]: Fixed in c787b59 — corrected the page metadata to 56 diagnostic checks / 10 structural / 46 semantic.

File: docs/validation.md [hunk 5]
```diff
@@ -224,6 +224,30 @@ error[DIP009]: duplicate edge
...
+Every edge `when` condition must parse into a valid expression. An unparseable
+condition leaves the edge's routing undefined, so the workflow cannot execute —
+it fails at `dippin simulate` and at runtime. The most common cause is using a
+tool-node field (like `marker_grep`) or an unknown operator in operator position.
+
+```
+error[DIP010]: edge A -> Z: invalid condition "marker_grep \"^ok\"": unknown operator "^ok"
+  --> pipeline.dip:14:5
+  = help: valid operators: = == != contains startswith endswith in (combine with and/or/not)
+```
```
[the remark this answers is not in the document]
Comment (clintecker) [reply docs/validation.md]: Fixed in c787b59 — added the `text` language tag to the DIP010 example fence.

File: docs/cli.md [hunk 6]
```diff
@@ -95,7 +95,7 @@ dippin parse pipeline.dip
 
 ### validate
 
-Run structural validation checks (DIP001–DIP009) on a workflow.
+Run structural validation checks (DIP001–DIP010) on a workflow.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply docs/cli.md:98]: Fixed in de6ec6e — `docs/cli.md` now reads "The 10 structural validation rules" (DIP010 is the 10th).

File: README.md [hunk 7]
```diff
@@ -30,7 +30,7 @@ graph LR
 | Shell scripts | `tool_command="#!/bin/sh\nset -eu\nif..."` | Real multiline, real syntax |
 | Model config | Untyped `llm_model="..."` attribute | Typed `model:` field with validation |
 | Branching | `condition="context.x!=y && context.a==b"` | `when ctx.x != "y" and ctx.a == "b"` |
-| Validation | Silent — typos in attrs are ignored | Diagnostic codes (DIP001–DIP009, DIP101–DIP137) |
+| Validation | Silent — typos in attrs are ignored | Diagnostic codes (DIP001–DIP010, DIP101–DIP137) |
```
[the remark this answers is not in the document]
Comment (clintecker) [reply README.md]: Fixed in de6ec6e — updated to DIP101–DIP146. This was pre-existing #99+ catalog drift; folded the full DIP146 sync into this PR per a maintainer decision rather than a separate docs-sync.

File: site/content/cli.md [hunk 8]
```diff
@@ -50,13 +50,13 @@ Bundle commands (`pack`, `unpack`, `inspect`) use a finer ladder so tooling can
...
   <div class="cmd-usage">dippin validate &lt;file&gt;</div>
-  <p>Run structural validation checks (DIP001-DIP009) on a workflow. Outputs "validation passed" or diagnostic messages. Exit code 1 if any errors found.</p>
+  <p>Run structural validation checks (DIP001-DIP010) on a workflow. Outputs "validation passed" or diagnostic messages. Exit code 1 if any errors found.</p>
 </div>
 
 <div class="cmd-card">
   <h3>lint</h3>
   <div class="cmd-usage">dippin lint [--extra-models &lt;spec&gt;] &lt;file&gt;</div>
-  <p>Run both structural validation and semantic linting (DIP001-DIP009 + DIP101-DIP134). All 40 diagnostic rules. Errors cause exit code 1; warnings alone exit 0.</p>
+  <p>Run both structural validation and semantic linting (DIP001-DIP010 + DIP101-DIP134). All 40 diagnostic rules. Errors cause exit code 1; warnings alone exit 0.</p>
```
[the remark this answers is not in the document]
Comment (clintecker) [reply site/content/cli.md]: Fixed in de6ec6e — now "DIP001-DIP010 + DIP101-DIP146. All 55 diagnostic rules".

File: docs/cli.md [hunk 9]
```diff
@@ -124,15 +124,15 @@ error[DIP003]: unknown node reference "InterpretX" in edge
 
 ### lint
 
-Run both structural validation and semantic linting (DIP001–DIP009 + DIP101–DIP145).
+Run both structural validation and semantic linting (DIP001–DIP010 + DIP101–DIP145).
```
[the remark this answers is not in the document]
Comment (clintecker) [reply docs/cli.md]: Fixed in de6ec6e — DIP101–DIP146.

File: docs/cli.md [hunk 10]
```diff
@@ -124,15 +124,15 @@ error[DIP003]: unknown node reference "InterpretX" in edge
...
 ```bash
 dippin lint [--extra-models <spec>] <file>
 ```
 
 **Input**: `.dip` or `.dot` file
 
-**Checks**: All 54 diagnostic rules. Errors (DIP001–DIP009) cause exit code 1. Warnings (DIP101–DIP145) are reported but don't affect the exit code.
+**Checks**: All 55 diagnostic rules. Errors (DIP001–DIP010) cause exit code 1. Warnings (DIP101–DIP145) are reported but don't affect the exit code.
 
 **Output**: All diagnostics (errors and warnings) to stderr.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply docs/cli.md]: Fixed in de6ec6e — the lint Checks line now reads DIP101–DIP146.

File: cmd/dippin/generated-spec.md [hunk 11]
```diff
@@ -560,7 +560,7 @@ Use `dippin help` (not `--help`) to see all commands.
 | Command | Purpose |
 |---------|---------|
 | `dippin parse <file>` | Output IR as JSON |
-| `dippin validate <file>` | Structural checks only (DIP001-DIP009) |
+| `dippin validate <file>` | Structural checks only (DIP001-DIP010) |
 | `dippin lint <file>` | Full validation + semantic warnings (DIP001–DIP145) |
```
[the remark this answers is not in the document]
Comment (clintecker) [reply cmd/dippin/generated-spec.md]: Fixed in de6ec6e — regenerated from the corrected `site/static/skill.md` input (not hand-edited); now DIP001–DIP146.

File: site/static/skill.md [hunk 12]
```diff
@@ -381,7 +381,7 @@ Use `dippin help` (not `--help`) to see all commands.
 | Command | Purpose |
 |---------|---------|
 | `dippin parse <file>` | Output IR as JSON |
-| `dippin validate <file>` | Structural checks only (DIP001-DIP009) |
+| `dippin validate <file>` | Structural checks only (DIP001-DIP010) |
 | `dippin lint <file>` | Full validation + semantic warnings (DIP001–DIP145) |
```
[the remark this answers is not in the document]
Comment (clintecker) [reply site/static/skill.md]: Fixed in de6ec6e — DIP001–DIP146 (this was the gen-spec input driving the stale generated-spec line above).

File: validator/diagnostic.go [hunk 13]
```diff
@@ -1,8 +1,9 @@
 // Package validator performs graph structure validation on Dippin IR workflows.
 //
-// It implements checks DIP001 through DIP009, covering structural correctness
+// It implements checks DIP001 through DIP010, covering structural correctness
 // of the workflow graph: start/exit existence, edge validity, reachability,
-// cycle detection, parallel/fan-in pairing, and duplicate detection.
+// cycle detection, parallel/fan-in pairing, duplicate detection, and edge
+// condition parseability.
 //
 // The validator is a pure IR consumer — it takes a *ir.Workflow and returns
```
[the remark this answers is not in the document]
Comment (clintecker) [reply validator/diagnostic.go]: Fixed in 2d2e398 — reworded the package doc to drop "pure IR consumer" and document that Validate lazily populates edge Condition.Parsed (idempotent) as a side effect, so callers should not assume the workflow is left unmutated.

File: README.md [hunk 14]
```diff
@@ -525,7 +525,7 @@ Everything flows through `ir.Workflow` — the canonical intermediate representa
 |---------|-------------|
 | `ir/` | Core types: `Workflow`, `Node`, `Edge`, `Condition` AST, typed `NodeConfig` sealed interface |
 | `parser/` | Indentation-aware lexer + recursive-descent parser producing IR |
-| `validator/` | 9 structural checks + 30 semantic lint rules |
+| `validator/` | 10 structural checks + 30 semantic lint rules |
```
[the remark this answers is not in the document]
Comment (clintecker) [reply README.md]: Fixed in 2d2e398 — the validator row now reads "10 structural checks + 46 semantic lint rules".

File: site/content/validation.md [hunk 15]
```diff
@@ -1,17 +1,17 @@
 ---
 title: "Validation & Linting"
-description: "40 diagnostic checks for AI pipeline workflows. 9 structural errors and 31 semantic warnings catch bugs before runtime."
+description: "56 diagnostic checks for AI pipeline workflows. 10 structural errors and 46 semantic warnings catch bugs before runtime."
 section_label: "Diagnostics"
-subtitle: "40 diagnostic checks — 9 structural errors and 31 semantic warnings — to catch problems before runtime."
+subtitle: "56 diagnostic checks — 10 structural errors and 46 semantic warnings — to catch problems before runtime."
```
[the remark this answers is not in the document]
Comment (clintecker) [reply site/content/validation.md]: Fixed in 2d2e398 — changed "56 diagnostic checks" to "56 diagnostic codes" in the front-matter (DIP138 is reserved/non-firing, so "codes" is the accurate term; also matches docs/validation.md).

File: validator/lint.go [hunk 16]
```diff
@@ -15,8 +15,17 @@ import (
...
+	// Ensure condition ASTs are populated — the AST-dependent lint checks
+	// (DIP103/120/121/122) read Condition.Parsed.
+	//
+	// parseEdgeConditions populates every parseable edge accumulate-all, so one
+	// unparseable edge does not mask those lints on later edges (the bug behind
+	// DIP010 / issue #98). Edge parse failures are reported as DIP010 from
+	// Validate; here we only need the population side effect. EnsureConditionsParsed
+	// then populates manager_loop node conditions; no lint reads node-condition
+	// Parsed today, so it is harmless that its edge walk re-fails and returns
+	// early when an edge is unparseable.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply validator/lint.go]: Fixed in 46c7b6a. Verified: `conditionPresent` is `c.Raw != "" || c.Parsed != nil`, so it does reference `Parsed` (you're right — the "no lint reads node-condition Parsed" wording was imprecise). But it's an OR with `Raw != ""`, which is true for any declared condition, so the result is Raw-dominated and leaving node `Parsed == nil` changes no lint result; and no manager_loop lint inspects a node condition's parsed AST structure (confirmed by grep). Reworded the comment to state exactly that: the early return leaves node conditions unparsed, but it changes no lint result because `conditionPresent` is Raw-dominated and nothing reads the node AST.

File: AGENTS.md [hunk 17]
```diff
@@ -60,8 +60,8 @@ Editor / agent integrations:
 
 ```sh
 dippin parse pipeline.dip          # JSON IR
-dippin validate pipeline.dip       # structural errors (DIP001-DIP009)
-dippin lint pipeline.dip           # semantic warnings (DIP101-DIP133)
+dippin validate pipeline.dip       # structural errors (DIP001-DIP010)
+dippin lint pipeline.dip           # semantic warnings (DIP101-DIP146)
```
[the remark this answers is not in the document]
Comment (clintecker) [reply AGENTS.md]: Fixed in 6177942 — reworded to "validation + semantic warnings" to reflect that `dippin lint` runs structural validation too and exits 1 on DIP001–DIP010 errors.

File: validator/lint.go [hunk 18]
```diff
@@ -15,8 +15,21 @@ import (
 //	structureResult := validator.Validate(w)
 //	lintResult := validator.Lint(w)
 func Lint(w *ir.Workflow) Result {
-	// Ensure condition ASTs are populated — the parser stores raw text
-	// but lint checks (DIP101, DIP102, DIP103) need parsed ASTs.
+	// Ensure condition ASTs are populated — the AST-dependent lint checks
+	// (DIP103/120/121/122) read edge Condition.Parsed.
+	//
+	// parseEdgeConditions populates every parseable edge accumulate-all, so one
+	// unparseable edge does not mask those lints on later edges (the bug behind
```
[the remark this answers is not in the document]
Comment (clintecker) [reply validator/lint.go:22]: The `Lint()` doc comment is correct as written. `validator.Lint()` emits DIP101–**DIP145**; DIP146 is emitted by the CLI-layer cross-file pass in `cmd/dippin` (the validator cannot read the child .dip), not by this function — see the DIP146 provenance note in `lint_codes.go` and the cross-file-pass comments in `lint_subgraph_tool_access.go`. So the function-level range is genuinely DIP145; changing it to DIP146 would wrongly imply `Lint()` emits it. The `dippin lint` *command* docs do say DIP101–DIP146, because the command runs that CLI pass — that distinction is intentional. Leaving as-is.

File: site/content/validation.md [hunk 19]
```diff
@@ -99,7 +99,7 @@ These must be fixed for a workflow to be valid. Each causes exit code 1.
   = help: remove the duplicate edge</pre>
 </div>
 
-## Semantic Warnings (DIP101-DIP134)
+## Semantic Warnings (DIP101-DIP146)
```
[the remark this answers is not in the document]
Comment (clintecker) [reply site/content/validation.md:102]: This is a curated highlights page — it shows the most common warnings as cards, not an exhaustive list (the same was true under the previous "DIP101–DIP134" heading; the range label denotes the category, not full coverage). The range itself is factually correct. I've left the heading as-is to avoid an editorial change to the published site copy in this bug-fix PR; happy to qualify it (e.g. "Semantic Warnings (DIP101–DIP146, selected)") if you'd prefer — your call.
